framework/platform/platform.cpp

```diff
@@ -369,7 +369,7 @@ bool Platform::start_app()
 
 	active_app = requested_app_info->create();
 
-	active_app->set_name(requested_app_info->id);
```

Comment:
We might want some method on AppInfo which returns the id if the name is empty?

Something like:

```
active_app->set_name(requested_app_info->human_readable_name());
```

Reply:
I think it would be better if the name in AppInfo was mandatory.

Reply:
I tend to agree that AppInfo should be mandatory.  We'll need to ensure all legacy samples are updated for it and it makes its way into acceptance criteria for new sample submissions.  An automated check would be helpful from our CI/CD